ishd.c

```diff
@@ -132,5 +132,7 @@ int main(void) {
         return 1;
     }
 
+    freeaddrinfo(addrs);
```

Comment:
Consider moving the `freeaddrinfo()` call right after the `bind()` call and its error handling; the addresses are not used after that. This line is only reached after the main server loop ends (if `accept4()` fails).